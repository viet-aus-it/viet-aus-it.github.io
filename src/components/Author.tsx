/* eslint-disable react/no-danger */
import { GatsbyImage } from 'gatsby-plugin-image';
import { rhythm } from '../utils/typography';

interface AuthorProps {
  author: ContentfulAuthor;
}

function Author({ author }: AuthorProps) {
  const avatar = author.image.gatsbyImageData;

  return (
    <div
      style={{
        display: `flex`,
        marginBottom: rhythm(2.5),
      }}
    >
      <GatsbyImage
        image={{
          ...avatar,
          layout: 'fixed',
          width: 150,
          height: 150,
        }}
        alt={author.name}
        style={{
          marginRight: rhythm(1 / 2),
          marginBottom: 0,
          minWidth: 50,
          borderRadius: `100%`,
        }}
      />
      <p>
        {`Written by `}
        <strong>{author.name}</strong>
        <span
          dangerouslySetInnerHTML={{
            __html: author.shortBio.childMarkdownRemark.html,
          }}
        />
      </p>
    </div>
  );
}

export default Author;
