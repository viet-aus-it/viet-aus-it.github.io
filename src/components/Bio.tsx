import { GatsbyImage } from 'gatsby-plugin-image';
import { rhythm } from '../utils/typography';
import useBioQuery from '../hooks/useBioQuery';

function Bio() {
  const { author, social, avatar } = useBioQuery();

  return (
    <div
      style={{
        display: `flex`,
        marginBottom: rhythm(2.5),
      }}
    >
      <GatsbyImage
        image={{
          ...avatar,
          width: 50,
          height: 50,
        }}
        alt={author.name}
        style={{
          marginRight: rhythm(1 / 2),
          marginBottom: 0,
          minWidth: 50,
          borderRadius: `100%`,
        }}
      />
      <p>
        {`A blog by `}
        <strong>{author.name}</strong>
        {` - ${author.summary}. We're on `}
        <a href={`https://github.com/${social.github}`}>GitHub</a>
        {`, `}
        <a href={`https://discord.gg/${social.discord}`}>Discord</a>
        {`. Our Facebook `}
        <a href={`https://discord.gg/${social.facebookGroup}`}>group</a>
        {` and `}
        <a href={`https://discord.gg/${social.facebookPage}`}>page</a>.
      </p>
    </div>
  );
}

export default Bio;
