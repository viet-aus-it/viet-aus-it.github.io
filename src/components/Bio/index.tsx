import { useStaticQuery, graphql } from 'gatsby';
import PureBio from './PureBio';

function Bio() {
  const data = useStaticQuery(graphql`
    query BioQuery {
      avatar: file(absolutePath: { regex: "/avatar.png/" }) {
        childImageSharp {
          gatsbyImageData(layout: FIXED)
        }
      }
      site {
        siteMetadata {
          author {
            name
            summary
          }
          social {
            github
          }
        }
      }
    }
  `);

  const { author, social } = data.site.siteMetadata;
  const avatar = data.avatar.childImageSharp.gatsbyImageData;

  return <PureBio author={author} social={social} avatar={avatar} />;
}

export default Bio;
