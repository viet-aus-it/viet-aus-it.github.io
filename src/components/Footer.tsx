import { FC } from 'react';

const Footer: FC = () => {
  const CurrentYear = new Date().getFullYear();

  return (
    <footer>
      {`Â© ${CurrentYear}, Built with `}
      <a href="https://www.gatsbyjs.org">Gatsby</a>
      {` Hosted on `}
      <a href="https://pages.github.com/">GitHub Pages</a>
    </footer>
  );
};

export default Footer;
