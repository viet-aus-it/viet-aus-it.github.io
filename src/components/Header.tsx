import { FC } from 'react';
import { Link } from 'gatsby';
import { rhythm, scale } from '../utils/typography';

interface HeaderProps {
  isRootPath: boolean;
  title: string;
}

const Header: FC<HeaderProps> = ({ isRootPath, title }) => {
  const LinkContent = (
    <Link
      style={{
        boxShadow: `none`,
        color: `inherit`,
      }}
      to="/"
    >
      {title}
    </Link>
  );

  const RootHeader = (
    <h1
      style={{
        ...scale(1.5),
        marginBottom: rhythm(1.5),
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h1>
  );

  const PageHeader = (
    <h3
      style={{
        fontFamily: `Montserrat, sans-serif`,
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h3>
  );

  const HeaderContent = isRootPath ? RootHeader : PageHeader;

  return <header>{HeaderContent}</header>;
};

export default Header;
