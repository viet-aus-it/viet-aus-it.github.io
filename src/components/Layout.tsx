import type { ReactNode } from 'react';
import Header from './Header';
import Footer from './Footer';
import { rhythm } from '../utils/typography';

interface LayoutProps {
  location: LocationType;
  title: string;
  children: ReactNode | Element;
}

function Layout({ location, title, children }: LayoutProps) {
  // This is not going to be undef, since it's going to be passed in upon build.
  // @ts-ignore
  const pathPrefix = __PATH_PREFIX__;
  const rootPath = `${pathPrefix}/`;
  const isRootPath = location.pathname === rootPath;

  return (
    <div
      style={{
        marginLeft: `auto`,
        marginRight: `auto`,
        maxWidth: rhythm(24),
        padding: `${rhythm(1.5)} ${rhythm(3 / 4)}`,
      }}
    >
      <Header title={title} isRootPath={isRootPath} />
      <main>{children}</main>
      <Footer />
    </div>
  );
}

export default Layout;
