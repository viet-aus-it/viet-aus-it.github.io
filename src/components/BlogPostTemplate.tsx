/* eslint-disable react/no-danger */
import { Link, graphql } from 'gatsby';
import { GatsbyImage } from 'gatsby-plugin-image';
import Author from './Author';
import Layout from './Layout';
import SEO from './SEO';
import { rhythm, scale } from '../utils/typography';
import formatPublishedDate from '../utils/formatPublishedDate';

interface BlogPostProps {
  data: { contentfulBlogPost: ContentfulPost };
  pageContext: {
    previous?: ContentfulPost;
    next?: ContentfulPost;
  };
  location: LocationType;
}

function BlogPostTemplate({ data, pageContext, location }: BlogPostProps) {
  const { previous, next } = pageContext;

  const post = data.contentfulBlogPost;
  const pageTitle = post.title;
  const pageDescription = post.description.childMarkdownRemark.excerpt;
  const publishedDate = formatPublishedDate(post.publishDate);
  const heroImage = post.heroImage.gatsbyImageData;
  const postAuthor = post.author;
  const bodyHTML = post.body.childMarkdownRemark.html;

  return (
    <Layout location={location} title={pageTitle}>
      <SEO title={pageTitle} description={pageDescription} />
      <article>
        <header>
          <h1
            style={{
              marginTop: rhythm(1),
              marginBottom: 0,
            }}
          >
            {pageTitle}
          </h1>
          <p
            style={{
              ...scale(-1 / 5),
              display: `block`,
              marginBottom: rhythm(1),
            }}
          >
            {publishedDate}
          </p>
        </header>
        <GatsbyImage image={heroImage} alt={pageTitle} />
        <section
          dangerouslySetInnerHTML={{
            __html: bodyHTML ?? '',
          }}
        />
        <hr style={{ marginBottom: rhythm(1) }} />
        <footer>
          <Author author={postAuthor} />
        </footer>
      </article>

      <nav>
        <ul
          style={{
            display: `flex`,
            flexWrap: `wrap`,
            justifyContent: `space-between`,
            listStyle: `none`,
            padding: 0,
          }}
        >
          <li>
            {previous && (
              <Link to={previous.slug} rel="prev">
                {`← ${previous.title}`}
              </Link>
            )}
          </li>
          <li>
            {next && (
              <Link to={next.slug} rel="next">
                {`${next.title} →`}
              </Link>
            )}
          </li>
        </ul>
      </nav>
    </Layout>
  );
}

export default BlogPostTemplate;

export const pageQuery = graphql`
  query BlogPostBySlug($slug: String!) {
    contentfulBlogPost(slug: { eq: $slug }) {
      title
      publishDate
      heroImage {
        gatsbyImageData
      }
      description {
        childMarkdownRemark {
          excerpt
        }
      }
      body {
        childMarkdownRemark {
          html
        }
      }
      author {
        name
        title
        company
        shortBio {
          shortBio
          childMarkdownRemark {
            html
          }
        }
        github
        facebook
        # twitter
        image {
          gatsbyImageData
        }
      }
    }
  }
`;
