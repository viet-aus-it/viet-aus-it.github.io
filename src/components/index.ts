import Author from './Author';
import Bio from './Bio';
import BlogPostTemplate from './BlogPostTemplate';
import Footer from './Footer';
import Header from './Header';
import Layout from './Layout';
import PostView from './PostView';
import SEO from './SEO';

export { Author, Bio, BlogPostTemplate, Footer, Header, Layout, PostView, SEO };
