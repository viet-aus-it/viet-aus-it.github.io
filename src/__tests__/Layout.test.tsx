/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';
import { Layout } from '../components';

describe('Layout component', () => {
  const title = 'Test Page';
  const children = <p>This is a test element</p>;

  it('Should render root path component', () => {
    const tree = render(
      <Layout title={title} location={{ pathname: '' }}>
        {children}
      </Layout>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should render other pages', () => {
    const tree = render(
      <Layout title={title} location={{ pathname: 'test-page' }}>
        {children}
      </Layout>
    );
    expect(tree).toMatchSnapshot();
  });
});
