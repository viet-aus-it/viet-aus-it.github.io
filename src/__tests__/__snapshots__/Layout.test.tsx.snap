// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Should render other pages 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-left: auto; margin-right: auto; max-width: 42rem; padding: 2.625rem 1.3125rem;"
      >
        <header>
          <h3
            style="font-family: Montserrat, sans-serif; margin-top: 0px;"
          >
            <a
              href="/"
              style="box-shadow: none;"
            >
              Test Page
            </a>
          </h3>
        </header>
        <main>
          <p>
            This is a test element
          </p>
        </main>
        <footer>
          © 2021, Built with 
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
           Hosted on 
          <a
            href="https://pages.github.com/"
          >
            GitHub Pages
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-left: auto; margin-right: auto; max-width: 42rem; padding: 2.625rem 1.3125rem;"
    >
      <header>
        <h3
          style="font-family: Montserrat, sans-serif; margin-top: 0px;"
        >
          <a
            href="/"
            style="box-shadow: none;"
          >
            Test Page
          </a>
        </h3>
      </header>
      <main>
        <p>
          This is a test element
        </p>
      </main>
      <footer>
        © 2021, Built with 
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
         Hosted on 
        <a
          href="https://pages.github.com/"
        >
          GitHub Pages
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Layout component Should render root path component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-left: auto; margin-right: auto; max-width: 42rem; padding: 2.625rem 1.3125rem;"
      >
        <header>
          <h3
            style="font-family: Montserrat, sans-serif; margin-top: 0px;"
          >
            <a
              href="/"
              style="box-shadow: none;"
            >
              Test Page
            </a>
          </h3>
        </header>
        <main>
          <p>
            This is a test element
          </p>
        </main>
        <footer>
          © 2021, Built with 
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
           Hosted on 
          <a
            href="https://pages.github.com/"
          >
            GitHub Pages
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-left: auto; margin-right: auto; max-width: 42rem; padding: 2.625rem 1.3125rem;"
    >
      <header>
        <h3
          style="font-family: Montserrat, sans-serif; margin-top: 0px;"
        >
          <a
            href="/"
            style="box-shadow: none;"
          >
            Test Page
          </a>
        </h3>
      </header>
      <main>
        <p>
          This is a test element
        </p>
      </main>
      <footer>
        © 2021, Built with 
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
         Hosted on 
        <a
          href="https://pages.github.com/"
        >
          GitHub Pages
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
